component/src/main/java/io/siddhi/extension/io/file/listeners/FileSystemListener.java

```diff
@@ -86,29 +87,18 @@ public boolean onMessage(RemoteFileSystemBaseMessage remoteFileSystemBaseEvent)
                     properties.put(Constants.FILE_READ_WAIT_TIMEOUT_KEY,
                             fileSourceConfiguration.getFileReadWaitTimeout());
                     properties.put(Constants.MODE, mode);
+                    properties.put(Constants.CRON_EXPRESSION, fileSourceConfiguration.getCronExpression());
                     VFSClientConnectorCallback carbonCallback = new VFSClientConnectorCallback();
-                    BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(
-                            ByteBuffer.wrap(fileURI.getBytes(StandardCharsets.UTF_8)), true);
-                    try {
-                        vfsClientConnector.send(carbonMessage, carbonCallback, properties);
-                        try {
-                            carbonCallback.waitTillDone(fileSourceConfiguration.getTimeout(), fileURI);
-                        } catch (InterruptedException e) {
-                            log.error(String.format("Failed to wait until file '%s' is processed.", fileURI), e);
-                            return false;
-                        }
-                        if (!actionAfterProcess.equalsIgnoreCase(Constants.KEEP)) {
-                            reProcessFile(vfsClientConnector, carbonCallback, properties, fileURI);
-                        }
-                    } catch (ClientConnectorException e) {
-                        log.error(String.format("Failed to provide file '%s' for consuming.", fileURI), e);
-                        carbonCallback.done(carbonMessage);
+                    if (fileSourceConfiguration.getCronExpression() != null) {
+                        FileCronListener.scheduleJob(fileSourceConfiguration, fileProcessor, vfsClientConnector);
```

Comment:
@Ajanthy when 'dir.uri' is given, whenever a file is added this method gets called. And according to your implementation, if two new files gets added, two new jobs will get scheduled.
So when the process starts when the cron fulfills, two processes will read the same file which may can affect the accuracy.